api/server/router/system/system_routes.go

```diff
@@ -67,24 +67,16 @@ func (s *systemRouter) getInfo(ctx context.Context, w http.ResponseWriter, r *ht
 	version := httputils.VersionFromContext(ctx)
 	if versions.LessThan(version, "1.25") {
 		// TODO: handle this conversion in engine-api
-		type oldInfo struct {
-			*types.Info
-			ExecutionDriver string
-		}
-		old := &oldInfo{
-			Info:            info,
-			ExecutionDriver: "<not supported>",
-		}
-		nameOnlySecurityOptions := []string{}
-		kvSecOpts, err := types.DecodeSecurityOptions(old.SecurityOptions)
+		kvSecOpts, err := types.DecodeSecurityOptions(info.SecurityOptions)
 		if err != nil {
-			return err
+			info.Warnings = append(info.Warnings, err.Error())
 		}
-		for _, s := range kvSecOpts {
-			nameOnlySecurityOptions = append(nameOnlySecurityOptions, s.Name)
+		var nameOnly []string
+		for _, so := range kvSecOpts {
+			nameOnly = append(nameOnly, so.Name)
 		}
-		old.SecurityOptions = nameOnlySecurityOptions
-		return httputils.WriteJSON(w, http.StatusOK, old)
```

Comment:
We don't return early now with `WriteJSON` now, so we'll also hit this branch:

https://github.com/moby/moby/blob/99a8e53573ebf6264a2bc74068532652fd97d4cb/api/server/router/system/system_routes.go#L89-L96

Reply:
Looks like not going through this branch previously was a bug and the current code seems to be a correct behavior - I assume this was intentional.

I assume that change was intentional, but pointing it out just in case 😅

Reply:
Ah, yes, this was intentional, but I should've called it out;

- KernelVersion was added in https://github.com/moby/moby/commit/6057e6ad70ba4550df358e89b9a7e1da61666604, which is part of Docker v0.5.1 (API v1.3); https://github.com/moby/moby/pull/1243
- KernelMemory was added in https://github.com/moby/moby/commit/f02381660819139839b07d900f432762fd5a3658, which is part of Docker v19.03.0-beta1 (API v1.40), but deprecated by the Kernel (and API v1.43). I guess technically, we should set it to `false` for older API versions, but we didn't do so before, so probably not an issue as _actual_ older clients won't know the field, so should ignore it.